### PR TITLE
Restructure packaging related files to better reflect usage.

### PR DESCRIPTION
**Pull request by Ferroin**
##### Summary

This is prep-work for eventual CPack RPM support.

The idea here is to group files used for packaging based on the packaging infrastructure they are used for and clearly organize them under a specific hierarchy within the CMake files.

This should be a non-semantic change, with no functional change in behavior for either the build process or end users.

##### Test Plan

CI passes on this PR.

### CODE REVIEW

File: packaging/cmake/Modules/Packaging.cmake [hunk 1]
```diff
@@ -9,6 +9,8 @@ set(CPACK_DEBIAN_DEBUGINFO_PACKAGE NO)
 
 set(CPACK_DEBIAN_PACKAGE_SHLIBDEPS YES)
 
+set(PKG_FILES "${CMAKE_SOURCE_DIR}/packaging/cmake/pkg-files")
```
Comment (vkalintiris) [inline packaging/cmake/Modules/Packaging.cmake:12]: This is not really needed, is it? I think can use a single definition from the main CMakeLists.txt file.

Comment (Ferroin) [reply packaging/cmake/Modules/Packaging.cmake:12]: It probably isn’t needed, I just kind of forgot it was there. In the the initial version of this change I had forgotten to modify the main CMakeLists file, and when I fixed that I forgot to remove this line in the packaging file. I’ll update the PR to exclude it.

File: packaging/cmake/Modules/Packaging.cmake [hunk 2]
```diff
@@ -93,10 +93,10 @@ endif()
 list(JOIN _main_deps ", " CPACK_DEBIAN_NETDATA_PACKAGE_DEPENDS)
 
 set(CPACK_DEBIAN_PACKAGE_CONTROL_EXTRA
-	  "${CMAKE_SOURCE_DIR}/packaging/cmake/control/netdata/conffiles;"
-	  "${CMAKE_SOURCE_DIR}/packaging/cmake/control/netdata/preinst"
-	  "${CMAKE_SOURCE_DIR}/packaging/cmake/control/netdata/postinst"
-	  "${CMAKE_SOURCE_DIR}/packaging/cmake/control/netdata/postrm")
+	  "${PKG_FILES}/deb/netdata/conffiles;"
```
Comment (vkalintiris) [inline packaging/cmake/Modules/Packaging.cmake:96]: @Ferroin You  forgot to rename these to PKG_FILES_PATH.

**Comment by Ferroin** [comment]
Rebased to resolve merge conflict.

File: packaging/cmake/Modules/Packaging.cmake (hunk 2 above)
Comment (Ferroin) [reply packaging/cmake/Modules/Packaging.cmake]: Should be resolved now (as well as the merge conflict).

**Review by vkalintiris — APPROVED** [review]

**Pull request merged**